### PR TITLE
#5 Create a method for getting user's feed messages

### PR DESCRIPTION
**Pull request by artemlisikh**
issue #5

### CODE REVIEW

File: src/test/java/com/geronimo/MessageServiceTest.java [hunk 1]
```diff
@@ -46,6 +47,31 @@ public void after() {
 
     @Test
     @Transactional
+    public void testListFeedMessagesServiceMethod() {
+        author.addFollower(author); //I'm following myself, why not?
```
Comment (olisikh) [inline src/test/java/com/geronimo/MessageServiceTest.java]: This should be prohibited, liking yourself is ok, but following, oh no, this is too dangerous ;(

Comment (artemlisikh) [reply src/test/java/com/geronimo/MessageServiceTest.java]: working on it

File: src/main/java/com/geronimo/service/UserService.java [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+
+    @Override
+    public List<User> getFollowingUsers(Long id) {
+        Validate.notNull(id);
+
+        return userRepository.getFollowingUsers(id);
+    }
+
+    @Override
+    public User getUser(String username) {
```
Comment (olisikh) [inline src/main/java/com/geronimo/service/UserService.java]: getUserByUsername?

File: src/main/java/com/geronimo/service/UserService.java [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+    @Override
+    public void deleteUser(User user) {
+        Validate.notNull(user);
+
+        userRepository.delete(user);
+    }
+
+    @Transactional
+    @Override
+    public void deleteUser(Long id) {
```
Comment (olisikh) [inline src/main/java/com/geronimo/service/UserService.java]: ById

File: src/main/java/com/geronimo/service/UserService.java [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+        Validate.notNull(id);
+
+        userRepository.delete(id);
+    }
+
+    @Transactional
+    @Override
+    public void deleteUser(String username) {
```
Comment (olisikh) [inline src/main/java/com/geronimo/service/UserService.java]: ByUsername

**Review by olisikh — COMMENTED** [review]
Converter looks good, but I can't check what's wrong as I'm going to bed :)

File: src/main/java/com/geronimo/service/UserService.java (hunk 2 above)
Comment (artemlisikh) [reply src/main/java/com/geronimo/service/UserService.java]: are you kidding me?

File: src/main/java/com/geronimo/service/UserService.java (hunk 3 above)
Comment (artemlisikh) [reply src/main/java/com/geronimo/service/UserService.java]: seriously?

File: src/main/java/com/geronimo/service/UserService.java (hunk 4 above)
Comment (artemlisikh) [reply src/main/java/com/geronimo/service/UserService.java]: you're not kidding me.

File: src/main/java/com/geronimo/model/Message.java [hunk 5]
```diff
@@ -21,6 +21,7 @@
     public Message(String text, User author) {
         this.text = text;
         this.author = author;
+        author.addMessage(this);
```
Comment (olisikh) [inline src/main/java/com/geronimo/model/Message.java]: Let's not modify 3rd party entity from a constructor of another entity, let's move this logic to the service, e.g. in postMessage we can add a message to an author if we really need to before saving.

File: src/test/java/com/geronimo/MessageServiceTest.java [hunk 6]
```diff
@@ -33,33 +33,30 @@
...
         entityManager.flush();
 
         author = userService.saveUser(new User("nice_user", "nice_password"));
+        badUser = userService.saveUser(new User("not_nice_user", "neither_nice_password"));
     }
 
     @Transactional
     @After
     public void after() {
+        userService.deleteUserById(badUser.getId());
```
Comment (olisikh) [inline src/test/java/com/geronimo/MessageServiceTest.java]: Right after the line deleteByUserId, deleteUser method is used. Looks a little bit weird :)

Why not to do this the same way?

**Review by olisikh — COMMENTED** [review]
Great change, but a couple of question added.

File: src/test/java/com/geronimo/MessageServiceTest.java (hunk 6 above)
Comment (konteno) [reply src/test/java/com/geronimo/MessageServiceTest.java]: Y O L O

**Review by artemlisikh — COMMENTED** [review]
Splendid work!

**Review by olisikh — APPROVED** [review]

**Pull request merged**